inst/include/Rcpp/String.h

```diff
@@ -64,10 +63,10 @@ namespace Rcpp {
             RCPP_STRING_DEBUG("String(const String&)");
         }
 
-        String(const String& other, const std::string& enc) : data(other.get_sexp()), valid(true), buffer_ready(false) {
+        String(const String& other, cetype_t enc) : data(other.get_sexp()), valid(true), buffer_ready(false) {
```

Comment:
I would be in favor of removing them -- I think it's better for users to be explicit with `set_encoding` if they wanted to force a different encoding.